lightning/src/ln/channel.rs

```diff
@@ -2100,6 +2100,45 @@ impl<SP: Deref> Channel<SP> where
 		Ok(())
 	}
 
+	/// The total fee paid to close the channel and claim an HTLC-success transaction, accounting
+	/// for an added input (spending a P2WPKH) and P2WPKH output for fee bumping.
+	fn cost_to_claim_onchain<L: Deref>(&self, feerate_per_kw: u32, logger: &L) -> u64
+	where L::Target: Logger
+	{
+		let commitment_weight = if self.context.is_outbound() {
+			let keys = self.context.build_holder_transaction_keys(self.context.cur_holder_commitment_transaction_number);
+			let commitment_stats = self.context.build_commitment_transaction(self.context.cur_holder_commitment_transaction_number, &keys, true, true, logger);
+			commitment_stats.total_fee_sat
+		} else { 0 };
+
+		let htlc_weight = htlc_success_tx_weight(self.context.get_channel_type());
+		let p2wpkh_weight = 31 * 4;
+		let input_weight = 272; // Including witness, assuming it spends from a p2wpkh
```

Comment:
By the way I got this number from [this post](https://bitcoin.stackexchange.com/questions/100159/what-is-the-size-and-weight-of-a-p2wpkh-input)